src/static/index.html

```diff
@@ -0,0 +1,133 @@
+<!DOCTYPE html>
+
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="description" content="" />
+    <meta name="author" content="" />
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Inter:wght@400;700&display=swap"
+      rel="stylesheet"
+    />
+    <link rel="stylesheet" href="./style.css" />
+  </head>
+
+  <body>
+    <div class="container">
+      <div class="title">Repo Report!</div>
+      <div class="instructions">
+        <div>Enter Command and press Enter!</div>
+        <div id="status" class="status-hide">
+          <div class="loader">
+            <div></div>
+            <div></div>
+            <div></div>
+            <div></div>
+          </div>
+          <div class="status-text">Getting Data from Github...</div>
+        </div>
+      </div>
+
+      <input class="command" type="text" name="command" id="command" />
+      <div class="table-container">
+        <table class="table" cellspacing="0" cellpadding="0">
+          <thead id="table-head" class="table-head"></thead>
+          <tbody id="table-body"></tbody>
+        </table>
+      </div>
+    </div>
+    <script src="https://unpkg.com/axios/dist/axios.min.js"></script>
+    <script>
+      const executeCommand = async (command) => {
+        const res = await axios.post("/command", {
+          command: command,
+        })
+        return res.data.output
+      }
+
+      const handleInput = async (event) => {
+        if (event.keyCode === 13) {
+          event.preventDefault()
+          showStatus()
+          clearTable()
+          const response = await executeCommand(input.value)
+          hidestatus()
+          renderHead(response.metrics)
+          renderBody(response.repos, response.rows)
+        }
+      }
+
+      const clearTable = () => {
+        const tableHead = document.getElementById("table-head")
+        const tableBody = document.getElementById("table-body")
+
+        while (tableHead.firstChild) {
+          tableHead.firstChild.remove()
+        }
+        while (tableBody.firstChild) {
+          tableBody.firstChild.remove()
+        }
```

Comment:
any reason this is better than `tableHead.innerHTML = ''; tableBody.innerHTML = '';`?

Reply:
1. Using `element.innerHTML` is a known security risk since a malicious actor with access to the string can insert any valid HTML code into the webpage: https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#security_considerations

2.  innerHTML is slower, I wrote a simple benchmark test to demonstrate this: https://www.measurethat.net/Benchmarks/Show/16453/1/innerhtml-vs-elementfirstchildremove

Reply:
There’s no security risk when it’s set to a hardcoded string, especially an empty one.

I don’t find microbenchmark differences particularly compelling - the difference between 2 and 4 million ops a second is “both are instantaneous, performance doesn’t matter” - so it’s really about what’s clearer.

Let’s leave this as-is for now, since a longer term resolution will be to use a framework instead of vanilla.